src/core_ct/core.py

```diff
@@ -282,6 +343,30 @@ def chunk(self, x1=0, y1=0, z1=0, x2=None, y2=None, z2=None) -> Core:
         new_core = Core(self.pixel_array[x1:x2, y1:y2, z1:z2], self.pixel_dimensions)
         return new_core
 
+
+    def filter(self, brightness_filter) -> Core:
```

Comment:
The lambda input should be typed, there's not an amazing way to do this but I found an example using the typing library like this:
```python
from typing import Callable
is_even: Callable[[int], bool] = lambda x: (x % 2 == 0)
```